static/js/domain-registration.js

```diff
@@ -0,0 +1,136 @@
+                const domainCanBeRegistered = await domainRegistration.checkIfDomainIsSafeAndAvailable(requestedDomain);
+                
+                if (domainCanBeRegistered) {
+                    domainRegistration.modal.open();
+                } else {
+                    // If the domain cannot be registered, submit the form to init an error message.
+                    console.log("Domain not available. Please try again");
+                    const domainRegistrationForm = document.getElementById("domainRegistration");
+                    domainRegistrationForm.submit();
+                }
+            }
+        },
+        modal: {
+            open: () => {
+                const modal = document.querySelector(".js-modal-domain-registration-confirmation");
+                modal.classList.add("is-visible");
+
+                const requestedDomain = document.querySelector(".js-subdomain-value");
+                const requestedDomainPreview = document.querySelectorAll(".js-modal-domain-registration-confirmation-domain-preview");
+                const requestedDomainPreviewEnding = document.querySelector(".js-modal-domain-registration-confirmation-domain-ending")
+                
+                // Preview the domain the user requested
+                requestedDomainPreview.forEach( preview => {
+                    preview.textContent = requestedDomain.value;
+                })
+                
+                // .mozmail.com
+                requestedDomainPreviewEnding.textContent = `.${requestedDomain.dataset.domain}`;
+
+                const modalCancel = document.querySelector(".js-modal-domain-registration-cancel");
+                modalCancel.addEventListener("click", domainRegistration.modal.close, false);
+
+                const modalSubmit = document.querySelector(".js-modal-domain-registration-submit");
+                modalSubmit.disabled = true;
+                modalSubmit.addEventListener("click", domainRegistration.modal.formSubmit, false);
+
+                const modalConfirmCheckbox = document.querySelector(".js-modal-domain-registration-confirmation-checkbox");
+                modalConfirmCheckbox.checked = false;
+                modalConfirmCheckbox.addEventListener("change", domainRegistration.canModalBeSubmitted, false);
+
+                // Close the modal if the user clicks outside the modal
+                modal.addEventListener("click", (e) => {
+                    if (e.target.classList && e.target.classList.contains("is-visible")) {
+                        domainRegistration.modal.close();
+                    }
+                });
+
+                // Close modal if the user clicks the Escape key
+                document.addEventListener("keydown", domainRegistration.modal.close, false);
+            },
+            close: (e) => {
+                if (e && e.key && (e.key !== "Escape")) {
+                    // Do nothing! 
+                    return;
+                }   
```

Comment:
This logic catches all keystrokes while the modal is open/active. If a user clicks the Escape key, it continues on to close the modal. Other wise it drops/ignores it here.